### PR TITLE
For #6202 - Fix autocomplete not being generated on first key press

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For [#6202](https://github.com/mozilla-mobile/android-components/issues/6202)

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks.
- [x] **Tests**: This PR passes previously written tests.
- [x] **Changelog**: This PR does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).
### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt [hunk 1]
```diff
@@ -602,58 +602,65 @@ open class InlineAutocompleteEditText @JvmOverloads constructor(
...
+                            afterNonAutocompleteText.length > beforeChangedTextNonAutocomplete.length)
 
-            autoCompletePrefixLength = textLength
+            var shouldAddAutocomplete: Boolean =
+                    (
+                            // Don't autocomplete if search query
+                            !afterNonAutocompleteText.contains(" ") &&
+                                    // ... or if user is hitting a backspace (the string is getting smaller)
+                                    !hasTextBeenRemoved && afterNonAutocompleteText.isNotEmpty()
+                            ) || (hasTextBeenAdded && !afterNonAutocompleteText.contains(" "))
```
Comment (rocketsroger) [inline components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt]: Can we rework this section? I think we can simplify this check?

**Comment by rocketsroger** [comment]
https://github.com/mozilla-mobile/fenix/issues/8560 is a Fenix issue.  A Android Component PR should be done against a A-C issue.  Can you please create an Android Component issue and update the commit message?

Also can you please squash your commits?  Here is a good link on how to do it https://github.com/wprig/wprig/wiki/How-to-squash-commits

This PR works well, thanks for helping!

**Comment by rocketsroger** [comment]
bors r=rocketsroger

**Comment by codrut-topliceanu** [comment]
Thanks. 
Added some tests. PR incoming.

**Pull request merged**